code/modules/antagonists/eldritch_cult/eldritch_antag.dm

```diff
@@ -0,0 +1,256 @@
+				protection += P.target
+				objectives += P
+
+	var/datum/objective/sacrifice_ecult/SE = new
+	SE.owner = owner
+	SE.update_explanation_text()
+	objectives += SE
+
+/datum/antagonist/heretic/apply_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/current = owner.current
+	if(mob_override)
+		current = mob_override
+	add_antag_hud(antag_hud_type, antag_hud_name, current)
+	handle_clown_mutation(current, mob_override ? null : "Knowledge described in the book allowed you to overcome your clownish nature, allowing you to use complex items effectively.")
+	current.faction |= "heretics"
+
+/datum/antagonist/heretic/remove_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/current = owner.current
+	if(mob_override)
+		current = mob_override
+	remove_antag_hud(antag_hud_type, antag_hud_name, current)
+	handle_clown_mutation(current, removing = FALSE)
+	current.faction -= "heretics"
+
+/datum/antagonist/heretic/get_admin_commands()
+	. = ..()
+	.["Equip"] = CALLBACK(src,.proc/equip_cultist)
+
+/datum/antagonist/heretic/roundend_report()
+	var/list/parts = list()
+
+	var/cultiewin = TRUE
+
+	parts += printplayer(owner)
+	parts += "<b>Sacrifices Made:</b> [total_sacrifices]"
+
+	if(length(objectives))
+		var/count = 1
+		for(var/o in objectives)
+			var/datum/objective/objective = o
+			if(objective.check_completion())
+				parts += "<b>Objective #[count]</b>: [objective.explanation_text] <span class='greentext'>Success!</b></span>"
+			else
+				parts += "<b>Objective #[count]</b>: [objective.explanation_text] <span class='redtext'>Fail.</span>"
+				cultiewin = FALSE
+			count++
+
+	if(cultiewin)
+		parts += "<span class='greentext'>The heretic was successful!</span>"
+	else
+		parts += "<span class='redtext'>The heretic has failed.</span>"
+
+	parts += "<b>Knowledge Researched:</b> "
+
+	var/list/knowledge_message = list()
+	var/list/knowledge = get_all_knowledge()
+	for(var/X in knowledge)
+		var/datum/eldritch_knowledge/EK = knowledge[X]
+		knowledge_message += "[EK.name]"
+	parts += knowledge_message.Join(", ")
+
+	return parts.Join("<br>")
+////////////////
+// Knowledge //
+////////////////
+
+/datum/antagonist/heretic/proc/gain_knowledge(datum/eldritch_knowledge/EK)
+	if(has_knowledge(EK))
+		return FALSE
+	var/datum/eldritch_knowledge/initialized_knowledge = new EK
+	researched_knowledge[initialized_knowledge.type] = initialized_knowledge
+	initialized_knowledge.on_gain(owner.current)
+	return TRUE
+
+/datum/antagonist/heretic/proc/get_researchable_knowledge()
+	var/list/researchable_knowledge = list()
+	var/list/banned_knowledge = list()
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK = researched_knowledge[X]
+		researchable_knowledge |= EK.next_knowledge
+		banned_knowledge |= EK.banned_knowledge
+		banned_knowledge |= EK.type
+	researchable_knowledge -= banned_knowledge
+	return researchable_knowledge
+
+/datum/antagonist/heretic/proc/has_knowledge(datum/eldritch_knowledge/wanted)
```

Comment:
Remove this proc, use get_knowledge for everything, alternatively remove get_knowledge and use has_knowledge for everything.